### PR TITLE
Remove image url from the config [B2B sample applicaiton]

### PR DESCRIPTION
**Pull request by Achintha444**
## Purpose
The main purpose of this PR is to remove the image URL as a parameter of the config.json.

## Approach
The first step of this PR is to remove the _CommonConfig.ManagementAPIConfig.ImageBaseUrl_. After that, I send a URI to Asgardeo as the image resource. For example,

- When creating an enterprise IdP.
  - "/libs/themes/default/assets/images/identity-providers/enterprise-idp-illustration.svg"

is sent to the identity server, and used a locally saved version of those icons to show in the identity provider's interface of the sample application.

### CODE REVIEW

File: b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/idpSection/otherComponents/confirmAddRemoveLoginFlowModal.tsx [hunk 1]
```diff
@@ -73,15 +76,35 @@ export default function ConfirmAddRemoveLoginFlowModal(prop) {
...
         if (response) {
             onSuccess();
-            successTypeDialog(toaster, "Success", "IIdentity Provider Remove from the Login Flow Successfully.");
+            successTypeDialog(toaster, "Success", "Identity Provider Remove from the Login Flow Successfully.");
         } else {
             errorTypeDialog(toaster, "Error Occured", "Error occured while removing the identity provider. Try again.");
         }
     };
 
+
```
Comment (thivi) [inline b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/idpSection/otherComponents/confirmAddRemoveLoginFlowModal.tsx:88]: Shall we remove this line?

File: b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts [hunk 2]
```diff
@@ -17,10 +17,30 @@
...
-import { ENTERPRISE_ID, GOOGLE_ID } from "@b2bsample/shared/util/util-common";
+import { EMPTY_STRING, ENTERPRISE_ID, GOOGLE_ID } from "@b2bsample/shared/util/util-common";
 import IdentityProviderDiscoveryUrl from "./identityProviderDiscoveryUrl";
 import IdentityProviderTemplateModel from "./identityProviderTemplateModel";
-import config from "../../../../../../../config.json";
+import enterpriseImage from "../../../../../ui/ui-assets/src/lib/images/enterprise.svg";
+import googleImage from "../../../../../ui/ui-assets/src/lib/images/google.svg";
+
+/**
+ *
```
Comment (thivi) [inline b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts:27]:
```suggestion
```

File: b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts [hunk 3]
```diff
@@ -17,10 +17,30 @@
...
 import IdentityProviderDiscoveryUrl from "./identityProviderDiscoveryUrl";
 import IdentityProviderTemplateModel from "./identityProviderTemplateModel";
-import config from "../../../../../../../config.json";
+import enterpriseImage from "../../../../../ui/ui-assets/src/lib/images/enterprise.svg";
+import googleImage from "../../../../../ui/ui-assets/src/lib/images/google.svg";
+
+/**
+ * 
+ * @param templateId - template id of the identity provider
+ * @returns - local image for the relevant identity provider
```
Comment (thivi) [inline b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts:29]:
```suggestion

 * @returns - local image for the relevant identity provider
```

File: b2b-sample/libs/shared/data-access/data-access-common-models-util/src/lib/model/user/userUtils.ts [hunk 4]
```diff
@@ -36,15 +37,33 @@ export function decodeUser(user: User) : InternalUser {
...
+ */
+export function setEmail(email: string) {
+    const regex = /^DEFAULT\//g;
+
+    email = email.replace(regex, "");
+
+    return email;
+}
+
+
```
Comment (thivi) [inline b2b-sample/libs/shared/data-access/data-access-common-models-util/src/lib/model/user/userUtils.ts:54]:
```suggestion
```

File: b2b-sample/libs/shared/ui/ui-components/src/lib/components/accordianItemHeaderComponent/accordianItemHeaderComponent.tsx [hunk 5]
```diff
@@ -30,19 +30,25 @@ import { AccordianItemHeaderComponentProps } from
...
                 >
-                    { title ? title.charAt(0) : "" }
+                    <>
+                        { imageSrc ?
+                            (<Image
+                                src={ imageSrc }
+                                alt="idp image"
+                                width={ 50 } />)
+                            : null }
+                        { title ? title.charAt(0) : null }
```
Comment (thivi) [inline b2b-sample/libs/shared/ui/ui-components/src/lib/components/accordianItemHeaderComponent/accordianItemHeaderComponent.tsx:50]: Shall we format this properly?

**Review by thivi — APPROVED** [review]

**Comment by lashinijay** [comment]
Shall we mention the related git issue in the PR description? Check your other PRs also.

File: b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts [hunk 6]
```diff
@@ -17,10 +17,30 @@
...
+ */
+export function getImageForTheIdentityProvider(templateId: string): string {
+    if (GOOGLE_ID === templateId) {
+
+        return googleImage;
+    }
+    if (ENTERPRISE_ID === templateId) {
+
+        return enterpriseImage;
+
```
Comment (lashinijay) [inline b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts:39]: Let's remove the new line after the return statement to make it consistent with other places.

File: b2b-sample/libs/shared/data-access/data-access-common-models-util/src/lib/model/user/userUtils.ts [hunk 7]
```diff
@@ -36,15 +37,32 @@ export function decodeUser(user: User) : InternalUser {
...
 /**
  * 
  * @param userName - user name
  * 
  * @returns set username.
  */
 export function setUsername(userName: string) {
-
-    return userName;
+    if(config.BusinessAppConfig.ManagementAPIConfig.UserStore.trim()===""){
```
Comment (lashinijay) [inline b2b-sample/libs/shared/data-access/data-access-common-models-util/src/lib/model/user/userUtils.ts:61]: Formatting issue
`if(config.BusinessAppConfig.ManagementAPIConfig.UserStore.trim()===""){` ->
`if (config.BusinessAppConfig.ManagementAPIConfig.UserStore.trim() === "") {`

File: b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts [hunk 8]
```diff
@@ -17,10 +17,30 @@
...
-import { ENTERPRISE_ID, GOOGLE_ID } from "@b2bsample/shared/util/util-common";
+import { EMPTY_STRING, ENTERPRISE_ID, GOOGLE_ID } from "@b2bsample/shared/util/util-common";
 import IdentityProviderDiscoveryUrl from "./identityProviderDiscoveryUrl";
 import IdentityProviderTemplateModel from "./identityProviderTemplateModel";
-import config from "../../../../../../../config.json";
+import enterpriseImage from "../../../../../ui/ui-assets/src/lib/images/enterprise.svg";
+import googleImage from "../../../../../ui/ui-assets/src/lib/images/google.svg";
+
+/**
+ * @param templateId - template id of the identity provider
```
Comment (ThaminduR) [inline b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts:27]: add a new line here to make it consistent.

Comment (Achintha444) [reply b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts:27]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/e238027e3ab758576ae189fd6ec3d1496abb6315

File: b2b-sample/libs/shared/data-access/data-access-common-models-util/src/lib/model/user/userUtils.ts (hunk 7 above)
Comment (Achintha444) [reply b2b-sample/libs/shared/data-access/data-access-common-models-util/src/lib/model/user/userUtils.ts:61]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/e238027e3ab758576ae189fd6ec3d1496abb6315

File: b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts (hunk 6 above)
Comment (Achintha444) [reply b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/identityProviderUtils.ts:39]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/e238027e3ab758576ae189fd6ec3d1496abb6315

**Pull request merged**